File: web/serve.js
#!/usr/bin/env node
/**
 * Simple HTTP server for testing Recipe WASM locally.
 * Serves files with proper MIME types for WASM.
 *
 * Usage: node serve.js [port]
 */

const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = process.argv[2] || 8080;
const ROOT_DIR = __dirname;

const MIME_TYPES = {
    '.html': 'text/html',
    '.js': 'application/javascript',
    '.wasm': 'application/wasm',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon'
};

const server = http.createServer((req, res) => {
    // Parse URL and handle root
    let filePath = req.url === '/' ? '/index.html' : req.url;
    filePath = path.join(ROOT_DIR, filePath);

    // Get file extension
    const ext = path.extname(filePath).toLowerCase();
    const contentType = MIME_TYPES[ext] || 'application/octet-stream';

    // Read and serve file
    fs.readFile(filePath, (err, content) => {
        if (err) {
            if (err.code === 'ENOENT') {
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 Not Found</h1>', 'utf-8');
            } else {
                res.writeHead(500);
                res.end(`Server Error: ${err.code}`, 'utf-8');
            }
        } else {
            res.writeHead(200, {
                'Content-Type': contentType,
                'Access-Control-Allow-Origin': '*'
            });
            res.end(content, 'utf-8');
        }
    });
});

server.listen(PORT, () => {
    console.log('✅ Recipe WASM test server running');
    console.log(`   URL: http://localhost:${PORT}`);
    console.log('   Press Ctrl+C to stop');
    console.log('');
});

File: web-old/serve.py
#!/usr/bin/env python3
"""
Simple HTTP server for testing Recipe WASM locally.
Serves files with proper MIME types for WASM.
"""

import http.server
import socketserver
import os

PORT = 8080

class WAMSHandler(http.server.SimpleHTTPRequestHandler):
    """HTTP handler with WASM MIME type support"""

    extensions_map = {
        **http.server.SimpleHTTPRequestHandler.extensions_map,
        '.wasm': 'application/wasm',
        '.js': 'application/javascript',
    }

    def end_headers(self):
        # Add CORS headers for local development
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        http.server.SimpleHTTPRequestHandler.end_headers(self)

if __name__ == '__main__':
    # Change to web directory
    os.chdir(os.path.dirname(os.path.abspath(__file__)))

    with socketserver.TCPServer(("", PORT), WAMSHandler) as httpd:
        print(f"✅ Recipe WASM test server running")
        print(f"   URL: http://localhost:{PORT}")
        print(f"   Press Ctrl+C to stop")
        print()
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\n\nServer stopped.")
